File: backend/src/main/kotlin/mymoney/domain/repository/SubscriptionRepository.kt
package mymoney.domain.repository

import mymoney.domain.model.Subscription
import java.util.UUID

interface SubscriptionRepository {
    suspend fun create(subscription: Subscription): Subscription
    suspend fun findById(id: UUID): Subscription?
    suspend fun list(familyId: UUID): List<Subscription>
    suspend fun update(subscription: Subscription): Subscription
    suspend fun softDelete(id: UUID)
}

File: mobile/lib/domain/repository/subscription_repository.dart
import '../model/subscription.dart';

abstract class SubscriptionRepository {
  Future<void> create(Subscription subscription);
  Future<Subscription?> findById(String id);
  Future<List<Subscription>> listByFamily(String familyId);
  Future<void> update(Subscription subscription);
  Future<void> softDelete(String id);
  Stream<List<Subscription>> watchByFamily(String familyId);
}
